### PR TITLE
Celo Mento Protocol trades

### PR DESCRIPTION
**Pull request by tomfutago**
Aim: add Mento Protocol to dex.trades

### CODE REVIEW

**Comment by jeff-dude** [comment]
thank you as always @tomfutago 🤝 

we'll let @Hosuke give this a review for ya

**Comment by Hosuke** [comment]
Given that this PR involves changes to the `dex.trades` macro architecture, I'll be reviewing its compatibility in detail.

**Comment by jeff-dude** [comment]
> Given that this PR involves changes to the `dex.trades` macro architecture, I'll be reviewing its compatibility in detail.

agreed. we are working internally on redesign of `dex.trades`, so let's pause this a bit until we finalize the uniswap example we're working on here #4533

**Comment by tomfutago** [comment]
Cool, no worries. I like what's been done under #4533, great work @Hosuke ! Once it's finalised I can update my code to follow the same pattern (when I'm back from holiday next week).

**Comment by jeff-dude** [comment]
apologies for the delay here, we will try to finalize the linked dex PR this week 🙏

File: models/mento/celo/mento_celo_pools_legacy.sql
Comment (Hosuke) [inline models/mento/celo/mento_celo_pools_legacy.sql:1]: We may remove all these `*_legacy.sql` files now.

File: models/mento/celo/mento_celo_pools.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{{
+  config(
+    tags = ['dunesql'],
+    schema = 'mento_celo',
+    alias = alias('pools'),
```
Comment (Hosuke) [inline models/mento/celo/mento_celo_pools.sql:5]: We can also drop `alias()` functions in all config blocks.
```suggestion
    alias = 'pools',
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @tomfutago,
There are some changes need to make to this PR.

Also PR #4533 is ready.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @tomfutago 

Clean and nice!

**Review by Hosuke — APPROVED** [review]

**Review by jeff-dude — COMMENTED** [review]
thank you for the PR 🙏 

a few questions:
- the current state here only feeds into the new `dex.trades_beta`, is that okay for the short term until it replaces `dex.trades` completely? or do you need to add here to include in existing `dex.trades` structure?
- what's the end goal for the `mento_celo_pools` spells?
  - one main reason i ask is due to usage of tokens erc20 metadata within the spells. you are materializing them at the most upstream level with this metadata, which is a pattern we're moving away from in new sector designs, especially since token metadata will soon flip to a source rather than ref (more to come on this!)
  - the goal will be to have all spells which read from them to be a view and/or the end of the dbt lineage, to avoid long lineages and full refresh dependencies all over in orchestration.

**Comment by tomfutago** [comment]
> * the current state here only feeds into the new `dex.trades_beta`, is that okay for the short term until it replaces `dex.trades` completely? or do you need to add here to include in existing `dex.trades` structure?

yes, adding to `dex.trades_beta` is fine for now

> * what's the end goal for the `mento_celo_pools` spells?
>   
>   * one main reason i ask is due to usage of tokens erc20 metadata within the spells. you are materializing them at the most upstream level with this metadata, which is a pattern we're moving away from in new sector designs, especially since token metadata will soon flip to a source rather than ref (more to come on this!)
>   * the goal will be to have all spells which read from them to be a view and/or the end of the dbt lineage, to avoid long lineages and full refresh dependencies all over in orchestration.

agreed! will remove tokens erc20 refs and bring pool models to follow `dex_pools` model

**Comment by tomfutago** [comment]
pools updated now

>   * one main reason i ask is due to usage of tokens erc20 metadata within the spells. you are materializing them at the most upstream level with this metadata, which is a pattern we're moving away from in new sector designs, especially since token metadata will soon flip to a source rather than ref (more to come on this!)

on that last bit - exciting stuff! but does it mean my #4762 doesn't have a chance to get in before new magic starts flowing?

File: models/mento/celo/mento_celo_pools.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  config(
+    schema = 'mento_celo',
+    alias = 'pools',
+    materialized = 'view'
+  )
+}}
+
+{% set pool_models = [
+  ref('mento_celo_pools_v1'),
+  ref('mento_celo_pools_v2')
```
Comment (jeff-dude) [inline models/mento/celo/mento_celo_pools.sql:11]: similar to `dex.trades` lineage, should these be named as below?
`mento_v1_celo_pools`
`mento_v2_celo_pools`

Comment (jeff-dude) [reply models/mento/celo/mento_celo_pools.sql:11]: would mean changes to file names, config settings, refs, etc etc

File: models/mento/celo/mento_celo_pools_v1.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    schema = 'mento_celo',
+    alias = 'pools_v1',
+    materialized = 'table'
+  )
+}}
+
+select
+  'celo' as blockchain,
+  'mento' as project,
+  'v1' as version,
```
Comment (jeff-dude) [inline models/mento/celo/mento_celo_pools_v1.sql:12]: can you do me a favor and double check the usage of `version` in other dex pool models? do they preface with `v` in front of the version number?

File: models/mento/celo/mento_celo_pools_v1.sql [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{{
+  config(
+    schema = 'mento_celo',
+    alias = 'pools_v1',
+    materialized = 'table'
```
Comment (jeff-dude) [inline models/mento/celo/mento_celo_pools_v1.sql:5]: any reason v1 is full table refresh each time, while v2 is incremental?

**Review by jeff-dude — COMMENTED** [review]
few comments below

also, this shouldn't happen in this PR, but seems like `dex.pools` is a nice candidate for a redesign like `dex.trades` -- should be a bit easier to do, on number of models alone

**Comment by jeff-dude** [comment]
> on that last bit - exciting stuff! but does it mean my #4762 doesn't have a chance to get in before new magic starts flowing?

nope, these will continue until that work is final. it's in a PoC phase at the moment. i'll get it merged soon 🤝

File: models/mento/celo/mento_celo_pools_v1.sql (hunk 3 above)
Comment (jeff-dude) [reply models/mento/celo/mento_celo_pools_v1.sql]: yeah, we can fix in later PR for others, no worries

**Comment by jeff-dude** [comment]
> @hildobby do you want to take it over as you were the one who created dex.pools originally?

no rush here. if not prioritized, dune team will get to it eventually. just focusing on most commonly used ones first

**Review by jeff-dude — APPROVED** [review]
looks good -- thank you!

**Comment by jeff-dude** [comment]
fyi @Hosuke 
after this is merged, you may want to pull in from main branch into your open PR for cleaning up seed tests on dex trades new lineage for the mento project here

**Pull request merged**